### ISSUE TITLE
VNC window size is small after restarting container

### ISSUE THREAD

**Issue by Saerith** [BioDepot/BioDepot-workflow-builder#11]
This happened after I rebuilt the container with new code as well. From the discussion last week, it sounds like a quicktime issue, with how they size windows.

**Comment by lhhunghimself** [comment]
The total canvas size shown by the browser is currently fixed to the size that was first encountered when the browser connects. This is something that we need to fix with the noVNC package - probably an extra button to allow the user to resize everything.

The small windows can be fixed by specifying minimum sizes but that should probably depend on the canvas size. With tabbed and scrolling windows - I think this will be less of a problem but the long term fix is to link minimum windows sizes to the canvas size.

**Comment by Saerith** [comment]
It seems that when I first launch bwb the canvas fills the browser window's available space (desired behavior), and subsequent launches show the background (VNC?) components, like the reboot and shutdown buttons (undesirable). This one sounds tricky based on our discussion! 

Is sizing a GUI container based on its children components's sizes something that fits quicktime's paradigm?

**Comment by lhhunghimself** [comment]
**TLDR** - enter localhost:6080 as URL not localhost:6080/vnc_auto.html to connect to Bwb container if you want a resize

The noVNC module does a resize the first time the browser connects by doing a redirect and then redirects to  localhost:6080/vnc_auto.html (assuming that localhost is the ip you are using and 6080 is the port you chose in the docker run command)
Your browser will keep this localhost:6080/vnc_auto.html in the search/url bar of your browser when you quit the Bwb container. If you use this URL to connect next time - no resizing will be done. It will either use the last size or the default size which is something very small. To trigger a resize you need to use enter localhost:6080 into the search bar (or bookmark it...)

We really should fix this at some time but it's not our code so I would have to figure out exactly what the code is doing at startup and how best to fix it.

I will add it to the FAQs in the docs until we fix it. This, the lack of a cut and paste into the browser, and the mapping of keystrokes so that ^W doesn't close the window in Chrome are the most annoying things that need to be fixed with the noVNC module.

**Comment by lhhunghimself** [comment]
Looking at the code - it checks the initial url - if it does not have a route defined it will call a javascript method to get the window size and then redirect to another url which will then be handled by a python script which will send the size to the framebuffer. There appears to be no way to resize the framebuffer afterwards

So - it *could* be that you have a pop-up blocker on that is blocking the javascript method. If that is the case it will size the window to the default size

Alternatively you can pass the redirect url directly with the size  
i.e.
```
localhost:6080/redirect.html?width=1707&height=872 
```
or whatever width and height you want.
